.github/workflows/release.yaml

```diff
@@ -27,15 +31,16 @@ jobs:
 
       - uses: actions/setup-go@v5
         with:
-          go-version-file: go.mod
+          go-version: ${{ steps.go-version.outputs.result }}
 
       - id: release
         run: make release
 
       - uses: actions/upload-artifact@v4
         with:
-          name: trust-manager-${{ env.VERSION }}.tgz
+          name: ${{ steps.release.outputs.RELEASE_HELM_CHART_NAME }}-${{ steps.release.outputs.RELEASE_HELM_CHART_VERSION }}.tgz
```

Comment:
This change is also based on https://github.com/cert-manager/approver-policy/blob/main/.github/workflows/release.yaml.